flink-table/flink-table-planner/src/test/scala/org/apache/flink/table/planner/plan/batch/sql/agg/OverAggregateTest.scala

```diff
@@ -47,6 +48,18 @@ class OverAggregateTest extends TableTestBase {
     util.verifyExecPlan("SELECT c, SUM(a) OVER (ORDER BY b) FROM MyTable")
   }
 
+  @Test
+  def testDenseRankOnOrder(): Unit = {
```

Comment:
The plan tests in batch mode appear not to be able to capture the modifications introduced by this PR, and I apologize for mentioning the unit test for the plan earlier. 

I've attempted this case, and it seems only possible to test it within an integration test in stream mode. I propose we add another testRankOnOver test specifically for testing the RANK() function. 

Additionally, there seems to be a potential risk of encountering a NPE in `AggFunctionFactory#createPercentRankAggFunction` (even though `PercentRank` is not currently supported in streaming mode, the code suggests such a risk). Maybe we'd better address this as well? 

// ---------- 
// Broadening the discussion a bit, perhaps we should consider fundamentally preventing the `orderKeyIndexes`, which isn't marked as Nullable, from being null, although this would require substantial changes 🤔. I agree that we can just avoid huge changes as demonstrated by this PR.

Reply:
>I've attempted this case, and it seems only possible to test it within an integration test in stream mode. I propose we can add another testRankOnOver test specifically for testing the RANK() function.

this I didn't get, there is already`testRankOnOver`, would you like to add another one?

>Maybe we'd better address this as well
yep, makes sense, I extracted the logic and reused it for §createPercentRankAggFunction` as well

Reply:
What I mean is to add a test `testRankOnOver` in` OverAggregateITCase. scala` to test the function `Rank()`. Because the UT tests about plan here cannot test the modified part (when I reversed the changes in `AggFunctionFactory.scala`, these tests still passed).

Reply:
Besides plans there are already a test in `OverAggregateITCase. scala` both for `RANK` and `DENSE_RANK`, that's way I have this question
https://github.com/apache/flink/pull/19797/files#diff-0fa625835219d2a3fcacb2fa8de5274ed6795f10b06035c6c9b68d11b700f9a8R198-R203

Reply:
Oh, I missed it. Pardon me...